### ISSUE TITLE
[Documentation] item-text/value/disabled types aren't explained

### ISSUE THREAD

**Issue by jasonrhaddix** [vuetifyjs/vuetify#5557]
### Versions and Environment
**Vuetify:** 1.3.7
**Vue:** 2.5.17
**Browsers:** Chrome 70.0.3538.77
**OS:** Mac OS 10.14.0

### Steps to reproduce
1. Use v-select to render a combobox
2. Pass in :items as Array
3. Add item-disabled prop with String or Array value ( example :item-disabled="['a']")

### Expected Behavior
The appropriate item in the select menu should be disabled.

### Actual Behavior
All items are still enabled and item-disabled prop seemingly does nothing.

### Reproduction Link
<a href="https://codepen.io/Jasonrhaddix/pen/yQOEyP" target="_blank">https://codepen.io/Jasonrhaddix/pen/yQOEyP</a>

### Other comments
This functionality wasn't working in my previous version either (1.2.4)

**Comment by jacekkarczmarczyk** [comment]
https://codepen.io/anon/pen/qQZgYW

**Comment by undefinederror** [comment]
@jacekkarczmarczyk 
Sorry but what kind of explanation is that?
I am now facing the same problem, and the copepen you shared shows an implementation that is nowhere to be found in documentation.

It looks like an ugly workaround more than anything.

Could you please explain why jasonrhaddix' implementation - as per docs - doesn't work, while yours - as per God know what - does?

Thank you

**Comment by KaelWD** [comment]
> jasonrhaddix' implementation - as per docs

There's no example of an array item-text afaik

> yours - as per God know what

as per how it was implemented in #3845. If an array is passed it's used as a path to a property (`['a', 'b']` is `'a.b'`), not a list of item values as @jasonrhaddix assumed.

**Comment by undefinederror** [comment]
Thank you @KaelWD for replying to this, despite the closed status.

Ok I think I understand now.
String or Array define a path on the current item object to return a Boolean which will be used to populate `disable` on the current `<option>`.

This is sooooo not clear from docs
![image](https://user-images.githubusercontent.com/8209966/64022326-3d61c180-cb36-11e9-92b8-f1715e56ff43.png)

Btw, default = `disabled`. Shouldn't it be default = `false`?

Also, and quite importantly,  I think it would be worth mentioning that there is no way to disable v-select items that are populated from an array of strings, as opposed to an Array of objects.

`item-disabled` = `<String>` or `<Array>` - a path on the current object - make no sense because the current item isn't an object, and `<Function>` is simply not invoked. 
No warnings, no error, it simply does nothing.

This is well confusing as per current docs. I understand that [#3845](https://github.com/vuetifyjs/vuetify/pull/3845) would give understanding of this behaviour but obviously users can't be expected to peruse each commit and pull requests or even issues. This should be already clear from docs IMHO.

**Comment by jacekkarczmarczyk** [comment]
> Btw, default = disabled. Shouldn't it be default = false?

No, default = disabled means that v-select will check for `item.disabled` to detect whether item is disabled or no
`item: [{ text: 'foo', disabled: true }, {text: 'foo', disabled: false}]` - foo is disabled, bar is not

**Comment by undefinederror** [comment]
Thank you @jasonrhaddix 
I find I now got a clear picture of the whole thing

Not exactly a walk in the park though 😄

**Comment by KaelWD** [comment]
> no way to disable v-select items that are populated from an array of strings

I think there's already a feature request for this, I can't seem to find it though. 
The feature was added by some random guy as part of a huge monolithic PR so it didn't really go through a proper review process. 

> This is well confusing as per current docs

Hence I reopened and tagged as docs

> Not exactly a walk in the park

Originally v-select only supported objects with this exact structure:
```
{
  text: string
  value: string
  disabled?: boolean
  avatar?: boolean
}
```

but then item-text/item-value were added as string props so custom object shapes could be used and it grew from there.
